go/vt/vtgate/executor.go

```diff
@@ -1495,6 +1499,21 @@ func checkLikeOpt(likeOpt string, colNames []string) (string, error) {
 	return "", nil
 }
 
+// isValidPayloadSize validates whether a query payload is above the
+// configured MaxPayloadSize threshold. The PayloadSizeExceeded will increment
+// if the payload size exceeds the warnPayloadSize.
+
+func isValidPayloadSize(query string) bool {
+	payloadSize := len(query)
+	if *maxPayloadSize > 0 && payloadSize > *maxPayloadSize {
+		return false
+	}
+	if *warnPayloadSize > 0 && payloadSize > *warnPayloadSize {
+		warnings.Add("PayloadSizeExceeded", 1)
```

Comment:
This is confusing. IIUC, there's an error threshold and a warning threshold, and the warning is added only if the warn threshold is exceeded.The warning name (`PayloadSizeExceeded`) sounds like it is the max that was exceeded.

Reply:
Changed name to `WarnPayloadSizeExceeded` for added clarity.